src/pheval/runners/runner.py

```diff
@@ -14,6 +22,95 @@ class PhEvalRunner(ABC):
     output_dir: Path
     config_file: Path
     version: str
+    directory_path = None
+    input_dir_config = None
+
+    def __post_init__(self):
+        self.input_dir_config = parse_input_dir_config(self.input_dir)
+
+    def _get_tool(self):
+        return self.input_dir_config.tool
+
+    def _get_phenotype_only(self):
+        return self.input_dir_config.phenotype_only
+
+    @property
+    def runner_version_dir(self):
+        return Path(self.output_dir).joinpath(f"{self._get_tool()}-{self.version}")
+
+    @runner_version_dir.setter
+    def runner_version_dir(self, directory_path):
+        self.directory_path = Path(directory_path)
+
+    @property
+    def runner_input_commands_dir(self):
+        return Path(self.output_dir).joinpath(
+            f"{self._get_tool()}-{self.version}/{RUNNER_INPUT_COMMANDS_DIR}"
```

Comment:
This is the last thing I am not sure about. Why isn't this simply the `--input-dir` value?

Reply:
I mean, the directory path is formatted correctly, but I thought the logic of the path creation should reside in the `Makefile`, not here..

Reply:
Is this conceptually like a temporary directory where the runner may choose to deposit intermediate files?

Reply:
I think it is something that can be put in the temporary directory, it's not an important runner output in terms of the results you want to get out of PhEval. It's just something that can allow you to run the exact same samples with the settings outside of PhEval. Normally I use the `prepare_batch_files` alone when I benchmark Exomiser on different environments. I can add a path to the `temp_dir` that has this directory, and then if the `--temp-dir` is specified in the CLI then the path is created there, otherwise in the output directory? What do you suggest?

Reply:
It's not so much to do with the input dir, more so with the test data dir. I also don't think it is a step that can be skipped as I don't know any tools that can take a phenopacket directory or vcf directory alone and produce the results from there.

For Phen2Gene, to run on the command line you would need to run this (outside of PhEval)
```
python3 /Users/yaseminbridges/phen2gene/Phen2Gene/phen2gene.py --manual HP:0000256 HP:0002059 HP:0100309 HP:0003150 HP:0001332 -out ../../../full_test_pheval_run/output/phen2gene/corpus1_results/phen2gene_results/ --name Abdul_Wahab-2016-GCDH-Patient_5 -d /Users/yaseminbridges/phen2gene/Phen2Gene/lib
```

In the runner implementation there is a single file that contains a command like this for all the phenopacket samples, i.e. there are 385 lines within the file that contain all the commands for running the corpus (385 samples within that corpus). So in the PhEval run, you create this file that has all the commands and then this is run `bash {path/to/tool_input_commands_dir/file.txt}`

So it is using a mixture of data from the input dir and testdata dir but I'm not sure if it's really preprocessing any data - it's just the actual commands needed to run the tool.

Reply:
> I think the idea I had as well is that there could be fields in the config.yaml within the tool_specific_configurations that could allow for I guess a more specific running of a software, i.e. extra parameters that can be used could be specified here if they do not always want to use it as a default

For sure.

>  I had just gone with the idea that anything output by pheval run goes in the output dir.

You are right, ok. Thanks for persisting and changing my mind. Since an output dir is already connected to a specific tool/version/corpus combination, you should be able to simply call it 

```
        return Path(self.output_dir).joinpath(f"{TOOL_INPUT_COMMANDS_DIR}"
```

Right? Or is there a point of adding `{self._get_tool()}-{self.version}/` in there?

Reply:
For sure, there are two different ways of going about it, I can either give it a general directory within the `output_dir`

```python
return Path(self.output_dir).joinpath(f"{TOOL_INPUT_COMMANDS_DIR}")
```
 and then the name of the file/prefix will need to be specific to the run, i.e. `exomiser-13.2.0-default-corpus1-default.txt` so that the `run` command can precisely locate the correct file containing the commands and only run the ones specific to that single run, to avoid running previous runs. This prefix is something that I can provide in the getters.

Alternatively, you can give it a specific directory:

```python
return Path(self.output_dir).joinpath(f"{self._get_tool()}-{self.version}/{TOOL_INPUT_COMMANDS_DIR}")
```

and a less specific file name `default-corpus1-default.txt`.

Or even more specific directory location...

Whichever you prefer I can do, the file just needs to be named something that can be identifiable by the runner to only run those commands and not previously generated files from other runs

Reply:
In the way I thought about it, `Path(self.output_dir)` should be a directory which stores the results of a _specific run_. So it already is clear which exact corpus, variant / version of tool etc was used (results.yml). Why would you need to name the output directory at all?

Reply:
That was my understanding, exactly!

Reply:
I had no idea! I can remove some of the getters that I have then as they are redundant. So the building of the output directory structure:

```
output
├── exomiser-13.2.0
│   ├── default-corpus1-default
│   │   ├── pheval_gene_results
│   │   ├── pheval_variant_results
│   │   └── results
│   └── runner_input_commands
```
 is going to be handled by the MakeFile? Or do I still need to build it before the prepare step in pheval run?

Reply:
The way I was thinking of it, the Makefile will create:

```
output
├── exomiser-13.2.0
│   ├── default-corpus1-default
```

And pheval will create will create the rest!

```
pheval_gene_results
 pheval_variant_results
 results
```

Continue to challenge me! I trust you more than myself :P

Reply:
No problem, thank you with bearing with me too!

Reply:
discussed with Jules and he is happy with the PR as is, but we have suggested some changes on the naming of the output directories as well as the paths/information that the pheval analysis command needs to know, see #90